### PR TITLE
Rename `X25519` to `Curve25519`

### LINKED ISSUE #299: Rename X25519 to Curve25519
I realized that in the past I didn't properly understand the difference between those two words.
The current X25519 implementation doesn't actually use the X25519 key exchange method, it's still using 3DH, so I believe that the appropriate term is Curve25519 instead.

Does that make sense?

### PR DESCRIPTION
**Pull request by daxpedda**
Fixes #299.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -84,7 +84,7 @@
 
 * Added CipherSuite API for specifying underlying primitives
 * Added support for specifying a slow password hashing function
-* Collapsed SignalKeyPair to X25519KeyPair
+* Collapsed SignalKeyPair to Curve25519KeyPair
```
Comment (kevinlewi) [inline CHANGELOG.md:87]: This can probably not be changed since it is historical.

File: src/key_exchange/group/curve25519.rs [hunk 2]
```diff
@@ -22,11 +22,11 @@ use subtle::ConstantTimeEq;
 use super::KeGroup;
 use crate::errors::InternalError;
 
-/// Implementation for X25519.
-pub struct X25519;
+/// Implementation for Curve25519.
+pub struct Curve25519;
 
 /// The implementation of such a subgroup for Ristretto
```
Comment (kevinlewi) [inline src/key_exchange/group/curve25519.rs:28]: `Ristretto` -> `Curve25519`

**Review by kevinlewi — APPROVED** [review]
Two minor nits, but otherwise looks good. Thank you!

**Review by kevinlewi — CHANGES_REQUESTED** [review]
(Setting to request changes)

File: CHANGELOG.md (hunk 1 above)
Comment (daxpedda) [reply CHANGELOG.md:87]: Yeah, that was a mistake.

File: src/key_exchange/group/curve25519.rs (hunk 2 above)
Comment (daxpedda) [reply src/key_exchange/group/curve25519.rs]: Done.

**Review by kevinlewi — APPROVED** [review]
Thanks!

**Pull request merged**